unime/src/lib/components/atoms/Button.svelte

```diff
@@ -17,17 +20,33 @@
 </script>
 
 <!-- TODO: does it make sense to pass in the trigger here? -->
-{#if trigger}
-  <button
-    use:melt={trigger}
-    class="h-[48px] w-full rounded-xl px-4 py-2 text-[13px]/[24px] font-medium disabled:opacity-50 {variant_classes}"
-    {disabled}
-    on:click={() => dispatch('click')}>{label}</button
-  >
+{#if !loading}
```

Comment:
I would put the if to check for the trigger outside and loading-if inside. And then conditionally render the spinner and disable the two types of buttons.

Reply:
Let's refactor the `Button` in a future refactoring session :)

Reply:
Sound good!